pkg/report/report.go

```diff
@@ -21,7 +23,36 @@ type ScanReportParser interface {
 	Parse(string) (*types.UpdateManifest, error)
 }
 
-func TryParseScanReport(file string) (*types.UpdateManifest, error) {
+func TryParseScanReport(file, scanner string) (*types.UpdateManifest, error) {
+
+	if scanner == "trivy" {
+		return defaultParseScanReport(file)
+	}
+	return customParseScanReport(file, scanner)
+}
+
+func customParseScanReport(file, scanner string) (*types.UpdateManifest, error) {
+
+	// Execute the plugin binary
+	cmd := "copa-" + scanner
+	scannerCommand := exec.Command(cmd, file)
+	// Capture the output
+	scannerOutput, err := scannerCommand.CombinedOutput()
+
+	if err != nil {
+		return nil, fmt.Errorf("error running scanner %s: %w", scanner, err)
+	}
+	// Convert the output to a UpdateManifest struct
+	var um types.UpdateManifest
+	if err := json.Unmarshal(scannerOutput, &um); err != nil {
```

Comment:
from the specs, we need to version our api so our plugins can communicate with copa with an API

here are the steps:
- for types, let's split this into 2 packages (`v1alpha1` and `unversioned`). unversioned is the internal api where we would convert everything to. Directory structure would look something like:
```shell
types
├── unversioned
│   └── types.go
└── v1alpha1
    ├── convert.go
    └── types.go
```

- copy `types.go` into each (since they are exactly same for now) and create a const with an apiVersion (this is what providers can use, example: https://github.com/project-copacetic/scanner-parser-template/blob/apiVersion/main.go#L41C23-L41C23)

```go
package v1alpha1
const APIVersion = "v1alpha1"
...
```

and for unversioned

```go
package unversioned
const APIVersion = "unversioned"
...
```
- make sure to update all internal types references to use `unversioned`.  Idea is that plugin communicate via API then copa can use it's own internal API.

- instead of just unmarshalling here, first we want to check what `apiVersion` we got from the plugin so we can convert later on. We can unmarshal the output to `map[string]interface{}{}` 

- then create a function to take this output and we can convert to unversioned/internal api. something like:
```go
func convertToUnversionedAPI(scannerOutput []byte, m map[string]interface{}) (*unversioned.UpdateManifest, error) {
	switch v := m["apiVersion"].(type) {
	case string:
		if v == "v1alpha1" {
			um, err := v1alpha1.Convert_v1alpha1_UpdateManifest_To_unversioned_UpdateManifest(scannerOutput)
...
<for other versions, we would error out as unsupported>
```

- We can define conversion func in the `types/unversioned/convert.go`, which is basically just unmarshalling to the `unversioned` since they are same at this time.

let me know if you have any questions. feel free to reach out over slack.